src/west/config.py

```diff
@@ -58,37 +80,65 @@ def read_config():
     '''
 
     # Gather (potential) configuration file paths
+    files = []
 
     # System-wide and user-specific
+    if config_file == ConfigFile.ALL or config_file == ConfigFile.SYSTEM:
+        files.append(ConfigFile.SYSTEM.value)
 
-    if platform.system() == 'Linux':
+    if config_file == ConfigFile.ALL and platform.system() == 'Linux':
         # Probably wouldn't hurt to check $XDG_CONFIG_HOME (defaults to
-        # ~/.config) on all systems. It's listed in git-config(1). People were
-        # iffy about it as of writing though.
-        files = ['/etc/westconfig',
-                 os.path.join(os.environ.get('XDG_CONFIG_HOME',
-                                             os.path.expanduser('~/.config')),
-                              'west', 'config')]
-
-    elif platform.system() == 'Darwin':  # Mac OS
-        # This was seen on a local machine ($(prefix) = /usr/local)
-        files = ['/usr/local/etc/westconfig']
-    elif platform.system() == 'Windows':
-        # Seen on a local machine
-        files = [os.path.expandvars('%PROGRAMDATA%\\west\\config')]
+        # ~/.config) on all systems. It's listed in git-config(1). People
+        # were iffy about it as of writing though.
+        files.append(os.path.join(os.environ.get(
+            'XDG_CONFIG_HOME',
+            os.path.expanduser('~/.config')),
+            'west', 'config'))
 
-    files.append(os.path.expanduser('~/.westconfig'))
+    if config_file == ConfigFile.ALL or config_file == ConfigFile.GLOBAL:
+        files.append(ConfigFile.GLOBAL.value)
 
     # Repository-specific
 
-    files.append(os.path.join(west_dir(), 'config'))
+    if config_file == ConfigFile.ALL or config_file == ConfigFile.LOCAL:
+        files.append(os.path.join(west_dir(), ConfigFile.LOCAL.value))
 
     #
     # Parse all existing configuration files
     #
     config.read(files, encoding='utf-8')
 
 
+def update_config(section, key, value, configfile=ConfigFile.LOCAL):
+    '''
+    Updates 'key' in section 'section' in ConfigFile, creating
+    'section' if it does not exist and write the file afterwards.
+
+    If value is None/empty, 'key' is left as-is.
+    '''
+    if not value:
+        return
+
+    if configfile == ConfigFile.ALL:
+        # Not possible to update ConfigFile.ALL, needs specific conf file here.
+        return
+
+    local_parser = configparser.ConfigParser()
```

Comment:
> We had discussed moving away from the standard library module when implementing `west config` because of these two major problems when writing config files:
> 
>     1. any comments users write in their files will be deleted
> 
This is correct. Comments are lost.
Comparing this to be able to use standard python ConfigParser, then I think this is acceptable trade-off.

>     2. order of sections and keys written by the user are not preserved
> 
actually they are.
I have tested this, and sections are not reordered when reading / writing.

Reply:
> This is correct. Comments are lost.
> Comparing this to be able to use standard python ConfigParser, then I think this is acceptable > trade-off.

I strongly disagree especially since alternatives exist which preserve comments. I write comments in basically all of my config files and would be quite upset if a tool which updates config files deleted them. Note `git config` preserves comments.

> actually they are.

OK, my mistake -- looks like I was using outdated information.

Reply:
> I strongly disagree especially since alternatives exist which preserve comments. I write comments in basically all of my config files and would be quite upset if a tool which updates config files deleted them.

Ok.
Personally for `git` conf files I have never added comments, just used `git config`.

So I basically considered to see if this was requested.
But seems there is already one person who strongly wants this ;)

So will take a look at the options.

Reply:
Switched to configobj.